presto-hive/src/main/java/io/prestosql/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -550,18 +550,21 @@ private static OptionalDouble getPartitionRowCount(String partitionName, Map<Str
     @VisibleForTesting
     static Optional<DoubleRange> calculateRangeForPartitioningKey(HiveColumnHandle column, Type type, List<HivePartition> partitions)
     {
-        if (!isRangeSupported(type)) {
-            return Optional.empty();
-        }
-
-        List<Double> values = partitions.stream()
+        List<OptionalDouble> convertedValues = partitions.stream()
                 .map(HivePartition::getKeys)
                 .map(keys -> keys.get(column))
                 .filter(value -> !value.isNull())
                 .map(NullableValue::getValue)
                 .map(value -> convertPartitionValueToDouble(type, value))
                 .collect(toImmutableList());
 
+        if (convertedValues.stream().noneMatch(OptionalDouble::isPresent)) {
```

Comment:
could you add a comment here? It seems that all or none value will be present.

Maybe change the condition to `!convertedValues.stream().allMatch(Optional::isPresent)`

Reply:
They all should be present or absent. I want to fail if some are present and some are not.